### PR TITLE
GH-48 - Switch CI to Gihtub Actions

### PR DESCRIPTION
**Pull request by coopergillan**
### What

Set up a Github Actions workflow to build and run tests for the project.

### Why

* travis-ci.org ~is migrating to travis.com soon~ has migrated to travis-ci.com
* Docker is limiting image pulls via current Travis builds
* Github Actions is built in to Github projects like runbook

Relates to #48

Opening this here to see if it will run the Github Actions workflow.

### CODE REVIEW

**Comment by coopergillan** [comment]
This was first attempted directly in `braintree/runbook` with https://github.com/braintree/runbook/pull/50.

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
```
Comment (hollabaq86) [inline .github/workflows/ruby.yml:12]: We may want to change this to a specific major version for safety but I think this is fine to start and see if the thing runs:tm:

File: .github/workflows/ruby.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        exclude:
+          - ruby-version: 2.3.8
+            gemfile: gemfiles/activesupport_6.gemfile
+          - ruby-version: 2.4.10
+            gemfile: gemfiles/activesupport_6.gemfile
+        gemfile:
+          - gemfiles/activesupport_5.gemfile
+          - gemfiles/activesupport_6.gemfile
+        ruby-version:
+          - 2.3.8
```
Comment (hollabaq86) [inline .github/workflows/ruby.yml:26]: Just curious, are these the ruby versions we were previously testing against? For context 2.6 is EOL in March this year, so I'm curious.

We could always open an issue to track in the runbook library to drop support for older versions and bump the major version of the gem.

**Review by hollabaq86 — COMMENTED** [review]
This looks great to me, based on reviewing the works of former SDK folks adding GH Actions to the iOS SDK. I'm comfortable trying this out and seeing what happens!

We can always revert it if it breaks

File: .github/workflows/ruby.yml (hunk 1 above)
Comment (coopergillan) [reply .github/workflows/ruby.yml:12]: Yep, I think that's a good idea. It looks like https://github.com/braintree/runbook/pull/50 resulted in some green builds, so we can start tweaking it some now.

File: .github/workflows/ruby.yml (hunk 2 above)
Comment (coopergillan) [reply .github/workflows/ruby.yml:26]: @hollabaq86 - thanks for the review. Yep, [these were what we were testing against](https://github.com/braintree/runbook/blob/6f4d226c605c117bc8da88a74d57e68aca4196c4/.travis.yml#L6-L9) before in Travis.

I do think we should work on dropping support for older versions. That's a great callout.

**Comment by coopergillan** [comment]
Closed in favor of https://github.com/braintree/runbook/pull/50.

**Pull request closed without merging**